code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -128,13 +139,43 @@ def generate_on_your_data_llm_chain(
             "chat_history": chat_history,
         }
 
+    def generate_source_documents(self, search_results):
```

Comment:
This method adds sas token placeholder for the documents in Storage so that the link from citation redirects to the correct blob storage

Reply:
Why not put that comment in as a remark for the `generate_source_documents` method?

Reply:
Moved it into a separate method as per below comment from Adam